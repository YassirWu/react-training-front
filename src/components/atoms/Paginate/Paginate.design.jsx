// @ts-nocheck
import React from 'react';
import t from 'prop-types';
import ReactPaginate from 'react-paginate';

import styles from './Paginate.module.scss';

const Paginate = ({ onPageChange, numberOfPages }) => (
  <ReactPaginate
    pageCount={numberOfPages}
    previousLabel="Précédent"
    nextLabel="Suivant"
    onPageChange={({ selected }) => onPageChange(selected)}
    containerClassName={styles.pagination}
    previousClassName={styles.pagination__previous}
    nextClassName={styles.pagination__next}
    disabledClassName={styles.pagination__disabled}
    pageClassName={styles.pagination__page}
    breakClassName={styles.pagination__page}
    activeClassName={styles.pagination__active}
  />
);

Paginate.propTypes = {
  onPageChange: t.func.isRequired,
  numberOfPages: t.number.isRequired,
};

export default Paginate;
