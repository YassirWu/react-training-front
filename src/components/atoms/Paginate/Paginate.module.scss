.pagination {
  display: flex;
  justify-content: center;

  &__previous {
    margin-right: 15px;
    cursor: pointer;
  }

  &__next {
    margin-left: 5px;
    cursor: pointer;
  }

  &__disabled {
    color: #b7b6b6;
    cursor: default;
  }

  &__page {
    margin-right: 10px;
    cursor: pointer;
  }

  &__active {
    color: #37474f;
    cursor: default;
    font-weight: 700;
  }
}