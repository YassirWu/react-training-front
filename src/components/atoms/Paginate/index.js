export { default } from './Paginate.design';
