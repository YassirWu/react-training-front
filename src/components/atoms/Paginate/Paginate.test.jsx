import React from 'react';
import { shallow } from 'enzyme';

import Paginate from './Paginate.design';

describe('<Paginate />', () => {
  test('should renders Paginate component', () => {
    const wrapper = shallow((
      <Paginate
        onPageChange={jest.fn()}
        numberOfPages={3}
      />
    ));

    expect(wrapper).toMatchSnapshot();
  });
});

