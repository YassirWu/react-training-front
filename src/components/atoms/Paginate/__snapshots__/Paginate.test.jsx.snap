// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginate /> should renders Paginate component 1`] = `ShallowWrapper {}`;
